### ISSUE TITLE
PCILEECH: Failed to connect to the device

### ISSUE THREAD

**Issue by Morph-Eng** [ufrisk/pcileech#102]
Hi, 
i had some issued with my new boards ( i get some LambdaConcept R01 FPGA-Boards) and would perfom these in my Linux Host System. First i build the Kernel-Driver and loaded into kernel ( the new dev in /dev/ft60x0 would be there); then i build the pcileech with the leechcore.so ...
then i try to connect to the boards with
`sudo` ./pcileech probe -> Nothing happend. 
next i try 
sudo ./pcileech dump -out test -> nothing

I switch my Host-System to Windows64, install the FTDI with the VID_040&PID_601, downloading the FTD3XXX.dll & lib and startet in a terminal like this:
pcileech.exe probe -device fpga
:Failed to connect to the device
also pcileech.exe probe

I try to flash one board again, but after flashing sccessful, nothing change..

Can anybody tell me, what is going wrong?
Thanks

**Comment by ufrisk** [comment]
1. ensure that the PCIeScreamer blinks when pressing button "TEST BTN" in the image below. This will tell if the PCIeScreamer is flashed or not.

![](https://gist.githubusercontent.com/ufrisk/c5ba7b360335a13bbac2515e5e7bb9d7/raw/e4e9ae4bf5fe0f723d6afe30703ac97df7e2c905/__gh_pciescreamer2.jpg)

2.
can you please try `sudo ./pcileech dump -out none -v -vv -device fpga` what output do you get?

---

Also please note that The PCIeScreamer R01 contains a hw issue that may affect performance/stability negatively.

**Comment by Morph-Eng** [comment]
Hi,
the LED3 is still on, LED1+2 is flashing, there is no change for pressing SW3? .. I would the *.bin File V3.2 for the R01 Version and reflash the board.
I try your code, but still there is nothing, no prompt after >Return<

**Comment by Morph-Eng** [comment]
Now i programmed it with a 3.0 Version; i got this output now:
sudo ./pcileech dump -out none -v -vv -device fpga

DEVICE: FPGA: ERROR: Unable to retrieve required Device PCIe ID [1,v3.0,0000]
PCILEECH: Failed to connect to the device.

**Comment by ufrisk** [comment]
if it's blinking all the time and there is no special reaction when pressing the button marked as "TEST BTN" in the picture above that means you failed to flash the PCIeScreamer with the PCILeech firmware. It's the V3.2 you should use. Also make sure you have powered the device when flashing either externally via barrel jack or via PCIe. The power the flash cable provides isn't sufficient.

**Comment by ufrisk** [comment]
[1,v3.0,0000] means you have contact with the device over USB, but no PCIe Link. (0000 = PCIe Link ID)

**Comment by Morph-Eng** [comment]
Ok, i would try tomorrow to flash the Version 3.2; No Link on the PCIe? the Board is in x16 Link ... i would try tomorrow a secound PC with a different Mainboard... thank for help :-)

**Comment by Morph-Eng** [comment]
Hi ufrisk,
i flash the board with the version V3.2, now the output is (1,V3.2,0000), i tryd with two different machines and with a linux-host and a windows-host ..??

**Comment by ufrisk** [comment]
This means communication between PCIeScreamerR01 and attacking computer running PCILeech is working fine. The PCIe link is however down.

try to insert the board while the computer is off, then power on/boot the computer. try to insert it into different slots in each computer, try both the x1 slots and the x16 slots, and if you have the chance to try different computers. Do not use any adapters or extension cables with the R01 version, insert it directly into the slots.

Unfortunately the R01 is quite unstable, it most often works if inserted directly into the PCIe slots, but on some computers the R01 is not working at all. I hope you can get it to work.

**Comment by Morph-Eng** [comment]
Hi,
i think we got it... we change some computers and slots; so on some Acer-Computers different years the card would not funktion. On the oldest computer this could not startup the Windows-System, on the newer one the card started with Windows, but the same output like no PCIe-Link. We check it with a Computer from HP, in this case we got a link and grab the memory sucessfuly. Is there a list of computers with funktion / no funktion?

Many Thanks for help!
